src/main/scala/org/broadinstitute/hail/Utils.scala

```diff
@@ -407,6 +408,21 @@ class RichRDDByteArray(val r: RDD[Array[Byte]]) extends AnyVal {
   }
 }
 
+class RichPairRDD[K, V](val r: RDD[(K, V)])
+                       (implicit kt: ClassTag[K], vt: ClassTag[V], ord: Ordering[K] = null) {
+
+  def leftOuterJoinDistinct[W](other: RDD[(K, W)]): RDD[(K, (V, Option[W]))] = leftOuterJoinDistinct(other, defaultPartitioner(r, other))
+
+  def leftOuterJoinDistinct[W](other: RDD[(K, W)], partitioner: Partitioner) = {
+    r.cogroup(other, partitioner).flatMapValues { pair =>
+      if (pair._2.isEmpty)
+        pair._1.iterator.map(v => (v, None))
+      else
+        for (v <- pair._1.iterator; w <- pair._2.iterator.take(1)) yield (v, Some(w))
```

Comment:
I think in the context of the leftOuterJoinDistinct code that makes sense. In practice, do we require variants to be unique?